magenta/models/rl_tuner/rl_tuner.py

```diff
@@ -0,0 +1,2052 @@
+                                            hparams=self.note_rnn_hparams,
+                                            note_rnn_type=self.note_rnn_type)
+
+      tf.logging.info('Initializing target q network')
+      self.target_q_network = note_rnn_loader.NoteRNNLoader(self.graph,
+                                                   'target_q_network',
+                                                   self.note_rnn_checkpoint_dir,
+                                                   self.midi_primer,
+                                                   training_file_list=
+                                                   self.training_file_list,
+                                                   softmax_within_graph=False,
+                                                   backup_checkpoint_file=
+                                                   self.backup_checkpoint_file,
+                                                   hparams=self.note_rnn_hparams,
+                                                   note_rnn_type=self.note_rnn_type)
+
+      tf.logging.info('Initializing reward network')
+      self.reward_rnn = note_rnn_loader.NoteRNNLoader(self.graph,
+                                             'reward_rnn',
+                                             self.note_rnn_checkpoint_dir,
+                                             self.midi_primer,
+                                             training_file_list=
+                                             self.training_file_list,
+                                             softmax_within_graph=False,
+                                             backup_checkpoint_file=
+                                             self.backup_checkpoint_file,
+                                             hparams=self.note_rnn_hparams,
+                                             note_rnn_type=self.note_rnn_type)
+
+      tf.logging.info('Q network cell: %s', self.q_network.cell)
+
+      # Add rest of variables to graph.
+      tf.logging.info('Adding RL graph variables')
+      self.build_graph()
+
+      # Prepare saver and session.
+      self.saver = tf.train.Saver()
+      self.session = tf.Session(graph=self.graph)
+      self.session.run(tf.initialize_all_variables())
+
+      # Initialize internal networks.
+      if restore_from_checkpoint:
+        self.q_network.initialize_and_restore(self.session)
+        self.target_q_network.initialize_and_restore(self.session)
+        self.reward_rnn.initialize_and_restore(self.session)
+
+        # Double check that the model was initialized from checkpoint properly.
+        reward_vars = self.reward_rnn.variables()
+        q_vars = self.q_network.variables()
+
+        reward1 = self.session.run(reward_vars[0])
+        q1 = self.session.run(q_vars[0])
+
+        if np.sum((q1 - reward1)**2) == 0.0:
+            print "\nSuccessfully initialized internal networks from checkpointed model!"
```

Comment:
rather than print statements, should use tf.logging statements. here and elsewhere